src/neptune_scale/core/components/aggregating_queue.py

```diff
@@ -119,6 +118,7 @@ def get(self) -> BatchedOperations:
                     logger.debug("Batch closed due to size limit %s", batch_bytes + element.metadata_size)
                     break
 
+                elements_in_batch += 1
```

Comment:
you could put the `elements_in_batch += 1` in line 133 (right after and outside this if-else block)
it'll make it harder for someone modifying this code in the future to repeat this mistake and add another break after `elements_in_batch += 1`

Reply:
Yes, but then the case in `if batch is None` makes it more convoluted, right?

Reply:
hmm, I think I'm changing this batching code to a large degree in my PR anyway so I guess we could continue there